libraries/rush-sdk/src/RushGlobalFolder.ts

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'path';
+import { FileSystem } from '@rushstack/node-core-library';
+import { trueCasePathSync } from 'true-case-path';
+
+/**
+ * This class provides global folders that are used for rush's internal install locations.
+ *
+ * @privateRemarks
+ * Keep this in sync with `RushGlobalFolder` in rush.
+ */
+class RushGlobalFolder {
+  public readonly path: string;
+  public readonly nodeSpecificPath: string;
+
+  public constructor() {
+    // Because RushGlobalFolder is used by the front-end VersionSelector before EnvironmentConfiguration
+    // is initialized, we need to read it using a special internal API.
+    let rushGlobalFolderOverride: string | undefined = undefined;
+    if (typeof process !== 'undefined') {
+      rushGlobalFolderOverride = getRushGlobalFolderOverride(process.env);
+    }
+    if (rushGlobalFolderOverride !== undefined) {
+      this.path = rushGlobalFolderOverride;
+    } else {
+      this.path = path.join(getHomeFolder(), '.rush');
+    }
+
+    const normalizedNodeVersion: string = process.version.match(/^[a-z0-9\-\.]+$/i)
+      ? process.version
+      : 'unknown-version';
+    this.nodeSpecificPath = path.join(this.path, `node-${normalizedNodeVersion}`);
+  }
+}
```

Comment:
Instead of copying this from `rush-lib`, why don't we deep-import this from rush-lib's `lib-esm` folder and bundle all of `rush-sdk`?

Reply:
Hi @iclanton , could you elaborate a little bit about this idea? 

Are you talking about

1. make `rush-sdk` dependency over `rush-lib`
2. introduce webpack.config.js to bundle `rush-sdk`
3. use webpack deep-import plugin to handle `rush-lib` imports

I know copying and syncing is bad. I just want to better understand your proposal.

Reply:
Yeah that's basically what I had in mind. `rush-sdk` already has a devDependency on `rush-lib`.